agnpy/spectra/tests/test_spectra.py

```diff
@@ -817,4 +827,63 @@ def test_compare_calculation_with_calculation_split_into_two(self):
             eval_2.evaluate(gammas, 1, gamma_min, gamma_max),
             0.001)
 
+    def test_total_number_of_electrons_should_stay_after_time_evolution(self):
+        R_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e48 * u.erg,
+            4 / 3 * np.pi * R_b ** 3,
+            p=2.8,
+            gamma_min=1e2,
+            gamma_max=1e7,
+            mass=m_e,
+        )
+
+        blob = Blob(R_b, Distance(1e27, unit=u.cm).z, 10, 10, 1 * u.G, n_e=n_e)
+        synch = Synchrotron(blob)
+        ssc = SynchrotronSelfCompton(blob)
+
+        time = 60 * u.s
+        intervals = 600
+
+        electronsBefore = n_e.integrate(n_e.gamma_min, n_e.gamma_max)
+
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, time, subintervals_count=intervals)
+        electronsAfterSync = blob.n_e.integrate(blob.n_e.gamma_min, blob.n_e.gamma_max)
+
+        blob.n_e = n_e
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate_thomson, time, subintervals_count=intervals)
+        electronsAfterSscTh = blob.n_e.integrate(blob.n_e.gamma_min, blob.n_e.gamma_max)
+
+        blob.n_e = n_e
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate, time, subintervals_count=intervals)
+        electronsAfterSsc = blob.n_e.integrate(blob.n_e.gamma_min, blob.n_e.gamma_max)
+
+        # total electron density before: 1298.1 cm-3, afterSync: 1297.6 cm-3, afterSscTh: 1297.1 cm-3, afterSsc: 1229.5 cm-3
+        assert u.isclose(electronsBefore, electronsAfterSync, rtol=0.005)
+        assert u.isclose(electronsBefore, electronsAfterSscTh, rtol=0.005)
+        assert u.isclose(electronsBefore, electronsAfterSsc, rtol=0.005) #failing below rtol=0.06
+
+    def test_ssc_thompson_limit(self):
+        R_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e48 * u.erg,
+            4 / 3 * np.pi * R_b ** 3,
+            p=2.8,
+            gamma_min=1e1,
+            gamma_max=1e5, #low energy limit
+            mass=m_e,
+        )
+
+        blob = Blob(R_b, Distance(1e27, unit=u.cm).z, 10, 10, 1 * u.G, n_e=n_e)
+        ssc = SynchrotronSelfCompton(blob)
+
+        time = 60 * u.s
+        intervals = 200
+
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate_thomson, time, subintervals_count=intervals)
+        n_e_thompson = deepcopy(blob.n_e)
+
+        blob.n_e = n_e
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate, time, subintervals_count=intervals)
```

Comment:
this effectively compares that the time_evolution gives the same with two different energy_loss_rate functions that should be close in this energy range. I think it would be better to simply compare the output of electron_energy_loss_rate and ..._thomson on a range of gammas without executing the time_evolution. On the other hand, a test that we were discussing was to compute this analytically (that is relatively simply for Thomson range) and introduce this comparison here.

Reply:
Fixed